### PR TITLE
Added type hints for WeightWindows class

### PR DESCRIPTION
**Pull request by christinacai123**

### CODE REVIEW

File: openmc/weight_windows.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
+
 from collections.abc import Iterable
 from numbers import Real, Integral
```
Comment (paulromano) [inline openmc/weight_windows.py:4]: Our Python code follows [PEP8](https://peps.python.org/pep-0008/), which recommends grouping imports into three groups: standard-library modules, third-party modules, and local imports. In your case, both `__future__` and `typing` are standard-library and should be in the first block:
```suggestion
from __future__ import annotations
from collections.abc import Iterable
from numbers import Real, Integral
from typing import Iterable, List
```

File: openmc/weight_windows.py [hunk 2]
```diff
@@ -11,6 +13,8 @@
 from ._xml import get_text
 from .mixin import IDManagerMixin
 
+from typing import Iterable
+
```
Comment (paulromano) [inline openmc/weight_windows.py:17]:
```suggestion
```

File: openmc/weight_windows.py [hunk 3]
```diff
@@ -216,13 +220,13 @@ def energy_bounds(self, bounds):
...
     @property
-    def num_energy_bins(self):
+    def num_energy_bins(self) -> int:
         if self.energy_bounds is None:
             raise ValueError('Energy bounds are not set')
         return self.energy_bounds.size - 1
 
     @property
-    def lower_ww_bounds(self):
+    def lower_ww_bounds(self) -> Iterable[Real]:
```
Comment (paulromano) [inline openmc/weight_windows.py:229]: When this attribute is set, it is converted to a numpy array, so this should probably be:
```suggestion
    def lower_ww_bounds(self) -> np.ndarray:
```

File: openmc/weight_windows.py [hunk 4]
```diff
@@ -241,7 +245,7 @@ def lower_ww_bounds(self, bounds):
         self._lower_ww_bounds = bounds
 
     @property
-    def upper_ww_bounds(self):
+    def upper_ww_bounds(self) -> Iterable[Real]:
```
Comment (paulromano) [inline openmc/weight_windows.py:248]:
```suggestion
    def upper_ww_bounds(self) -> np.ndarray:
```

File: openmc/weight_windows.py [hunk 5]
```diff
@@ -441,7 +445,7 @@ def from_hdf5(cls, group, meshes):
         )
 
 
-def wwinp_to_wws(path):
+def wwinp_to_wws(path) -> WeightWindows:
```
Comment (paulromano) [inline openmc/weight_windows.py:448]: This actually returns a list of weight windows:
```suggestion
def wwinp_to_wws(path) -> List[WeightWindows]:
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for contribution @christinacai123! (and thanks to @shimwell for helping guide some first-time contributors). For the suggested changes below, note that you can [batch them together as one commit](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes) directly in the web interface.

**Comment by shimwell** [comment]
Super to see you made it from discourse to GitHub so quickly. Nice one

Hit all the "add suggestion to batch" buttons on [this](https://github.com/openmc-dev/openmc/pull/2462/files) page as the next step

**Review by christinacai123 — COMMENTED** [review]
added suggestions to weight_windows.py

**Review by paulromano — APPROVED** [review]
Looks good to me! Will merge once CI passes.

**Pull request merged**